workflow/controller/operator.go

```diff
@@ -3001,3 +3026,137 @@ func (woc *wfOperationCtx) loadExecutionSpec() (wfv1.TemplateReferenceHolder, wf
 
 	return tmplRef, executionParameters, nil
 }
+
+// sum the resources duration from all pods associated with DAG or steps
+// templates
+func (woc *wfOperationCtx) maybeSumTemplateResourcesDuration(nodeName string, tmplCtx *templateresolution.Context, tmplRefHolder wfv1.TemplateReferenceHolder) error {
```

Comment:
this seems excessive, could you simple iterate over `/status/nodes` to create "ownership" ? @simster7 thoughts?

Reply:
Will take a look at this

Reply:
No worries. I was trying to iterate, but I couldn't find a way of filtering the pods that could give me a node and show all of the parent templates associated with that node. I went with the tree traversal to avoid naming collisions and handle nested DAGs and steps like [this example](https://github.com/argoproj/argo/blob/master/examples/dag-nested.yaml). If there's a simpler way to get all the parent templates for a node, let me know, and I'll patch up the PR tomorrow. Thanks @alexec and @simster7

Reply:
It'd be good a write an func that can compute the resource durations that is decoupled from the execution of the workflow. This can be unpredictable. If it could just take `wfv1.Nodes` and add the `resourceDuration` using children, then it would be reliable.

Reply:
Makes sense. Where should I invoke that function? Would `processNextItem` be a good place, or would you recommend somewhere else?

Reply:
I agree with Alex here, this does not seem to be the right approach.

Why not just have this feature be UI/CLI only? The UI/CLI could shift through the resources durations of all the nodes and add them client side. Not sure why this needs to be persisted in the CRD

Reply:
Sounds good. Thanks for the review @simster7. I'm happy to switch it to be CLI only for now. I have a hard time booting the UI on my machine due to JS heap issues.

Going off your comment, I was thinking about having the `argo get` command do the summing when it does a tree traversal of the node statuses there. Does that seem like a good approach to you?

Reply:
I actually have a differing opinion. By computing it and having it populated to each node, we don't have to duplicate the computation, and we make it available to other tools. 

I think the biggest challenge in the PR is the computation itself. Can this be computed by traversing the node graph for example?

Reply:
I like it. I'll try playing around with traversing the node statuses this weekend.